src/main/java/com/zendesk/maxwell/schema/MysqlSavedSchema.java

```diff
@@ -761,4 +754,152 @@ public static String getSchemaPositionSHA(Long serverID, Position position) {
 		);
 		return DigestUtils.shaHex(shaString);
 	}
+
+	public int getSchemaChainLength() {
+		return schemaChain.length();
+	}
+
+	public boolean compact(Connection connection, int chainMaxLength) throws SQLException {
+		if (chainMaxLength < 0) {
+			return false;
+		}
+		if (schemaID == null) {
+			throw new RuntimeException("Schema not saved yet.");
+		}
+		if (schemaChain.length() < chainMaxLength) {
+			return false;
+		}
+
+		connection.setAutoCommit(false);
+		try {
+			// save schema data
+			saveFullSchemaData(connection, schemaID);
+
+			// remove reference to base
+			try (PreparedStatement stmt = connection.prepareStatement(
+				"UPDATE `schemas` SET `base_schema_id` = NULL, `deltas` = NULL WHERE `id`= ?"
+			)) {
+				stmt.setLong(1, schemaID);
+				stmt.executeUpdate();
+			}
+
+			// clean up
+			StringBuilder inSQL = new StringBuilder();
+			for (int i = 0; i < schemaChain.length(); i++) {
+				if (inSQL.length() > 0) {
+					inSQL.append(", ");
+				}
+				inSQL.append("?");
+			}
```

Comment:
no need to go to all this trouble to use prepared statements.  Just build up the IN list of schema_ids once and then inject it directly into all the DELETE queries.